hist/histv7/inc/ROOT/RHistImpl.hxx

```diff
@@ -386,88 +762,347 @@ public:
+      BinArray_t localBins = {};
+      Internal::RVirtualBinsToLocalBins<NDIMS - 1, NDIMS, BinArray_t, decltype(fAxes)>()(localBins, fAxes, virtualBins);
+      return localBins;
+   }
 
-   /// Normalized axes access, converting from actual axis type to base class
-   const RAxisBase &GetAxis(int iAxis) const final { return *std::apply(Internal::GetAxisView<AXISCONFIG...>, fAxes)[iAxis]; }
+   /// Computes the global index of a certain bin on an `NDIMS`-dimensional histogram,
+   /// knowing the local per-axis bin indices as returned by calling `FindBin()` on each axis.
+   template <int NDIMS>
+   int ComputeGlobalBin(BinArray_t& local_bins) const {
+      // Get regular bins out of the way
+      if (std::all_of(local_bins.cbegin(), local_bins.cend(),
+                     [](int bin) { return bin >= 1; })) {
+         for (int bin = 0; bin < NDIMS; bin++)
+            local_bins[bin] -= 1;
+         return ComputeGlobalBinRaw<NDIMS>(local_bins, &ROOT::Experimental::RAxisBase::GetNBinsNoOver) + 1;
+      }
+
+      // Convert bin indices to a zero-based coordinate system where the underflow
+      // bin (if any) has coordinate 0 and the overflow bin (if any) has
+      // coordinate N-1, where N is the axis' total number of bins.
+      BinArray_t virtual_bins = LocalBinsToVirtualBins<NDIMS>(local_bins);
+
+      // Deduce what the global bin index would be in this coordinate system that
+      // unifies regular and overflow bins.
+      const int global_virtual_bin = ComputeGlobalBinRaw<NDIMS>(virtual_bins, &ROOT::Experimental::RAxisBase::GetNBins);
+
+      // Move to 1-based and negative indexing
+      const int neg_1based_virtual_bin = -global_virtual_bin - 1;
+
+      // At this point, we have an index that represents a count of all bins, both
+      // regular and overflow, that are located before the current bin when
+      // enumerating histogram bins in row-major order.
+      //
+      // We will next count the number of _regular_ bins which are located before
+      // the current bin, and by removing this offset from the above index, we
+      // will get a count of overflow bins that are located before the current bin
+      // in row-major order. Which is what we want as our overflow bin index.
+      //
+      int total_regular_bins_before = 0;
+
+      // First, we need to know how many regular bins we leave behind us for each
+      // step on each axis, assuming that the bin from which we come was regular.
+      //
+      // If mathematically inclined, you can also think of this as the size of an
+      // hyperplane of regular bins when projecting on lower-numbered dimensions.
+      // See the docs of ComputeLocalBins for more on this worldview.
+      //
+      BinArray_t bin_sizes;
+      bin_sizes[0] = 1;
+      Internal::RGetNRegularBinsBefore<NDIMS - 2, NDIMS, BinArray_t, decltype(fAxes)>()(bin_sizes, fAxes);
+
+      // Then, starting from the _last_ histogram dimension...
```

Comment:
Now that most of the loop has been extracted to `RComputeGlobalBin`, this comment feels a little too full of suspense.

```suggestion
      // With that, we can deduce how many regular bins lie before us.
```